packages/ng/forms/form-field-id.directive.ts

```diff
@@ -0,0 +1,42 @@
+import { Directive, HostBinding, inject, Input } from '@angular/core';
+import { FormFieldComponent } from '../form-field/form-field.component';
+import { filter, first } from 'rxjs/operators';
+
+@Directive({
+	selector: '[luFormFieldId]',
+	standalone: true,
+})
+export class FormFieldIdDirective {
+	#formFieldComponent = inject(FormFieldComponent);
+
+	#suffix: string;
+
+	@Input({
+		required: true,
+		alias: 'luFormFieldId',
+	})
+	set suffix(suffix: string) {
+		this.#suffix = suffix;
+		if (this.#formFieldComponent.ready) {
+			this.applyLabelledBy();
+		}
+	}
+
+	@Input()
+	labelledByStrategy: 'prepend' | 'append' = 'append';
+
+	@HostBinding('attr.id')
+	get id(): string {
+		return `${this.#formFieldComponent.id}-${this.#suffix}`;
+	}
+
+	constructor() {
+		this.#formFieldComponent.ready$.pipe(filter(Boolean), first()).subscribe(() => {
```

Comment:
I would play more defensively with a `take(1)`. If the component is created for a very short time, `first` can throw because `ready$` is completed before sending a `true`.

Reply:
TIL !